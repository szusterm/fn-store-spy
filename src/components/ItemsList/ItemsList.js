import React, {Component} from 'react';
import PropTypes from 'prop-types';

import ItemCard from '../ItemCard';

import './styles.scss';

class ItemsList extends Component {
	render() {
		const {items} = this.props;

		return (
			<div>
				{
					items.map((item) => {
						const {key, name, price, rarity, imgSrc} = item;

						return (
							<ItemCard
								key={key}
								name={name}
								price={price}
								rarity={rarity}
								imgSrc={imgSrc}
							/>
						);
					})
				}
			</div>
		);
	}
}

ItemsList.propTypes = {
	items: PropTypes.array
};

export default ItemsList;