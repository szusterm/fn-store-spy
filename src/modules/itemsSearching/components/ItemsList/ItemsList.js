import React, {Component} from 'react';
import PropTypes from 'prop-types';

import ItemCard from '../ItemSearchCard';

import './styles.scss';

export class ItemsList extends Component {
	render() {
		const {items} = this.props;

		return (
			<div className={'col-12 col-sm-12 col-md-11 col-lg-10 col-xl-8'}>
				<div className={'container-fluid'}>
					<div className={'row no-gutter'}>
						{
							items.map((item) => {
								const {_id, fnbrId, name, price, rarity, type, imageSrc} = item;

								return (
									<ItemCard
										key={_id}
										fnbrId={fnbrId}
										name={name}
										price={price}
										rarity={rarity}
										type={type}
										imageSrc={imageSrc}
									/>
								);
							})
						}
					</div>
				</div>
			</div>
		);
	}
}

ItemsList.propTypes = {
	items: PropTypes.array
};

export default ItemsList;