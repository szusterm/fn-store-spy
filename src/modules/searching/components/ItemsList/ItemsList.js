import React, {Component} from 'react';
import PropTypes from 'prop-types';

import ItemCard from '../ItemSearchCard/index';

import './styles.scss';

export class ItemsList extends Component {
	render() {
		const {items} = this.props;

		return (
			<div className={'container-fluid'}>
				<div className={'row no-gutter'}>
					{
						items.map((item) => {
							const {_id, name, price, rarity, type, imageSrc} = item;

							return (
								<div
									key={_id}
									className={'col-6 col-sm-6 col-md-4 col-lg-3 col-xl-3'}
								>
									<ItemCard
										_id={_id}
										name={name}
										price={price}
										rarity={rarity}
										type={type}
										imageSrc={imageSrc}
									/>
								</div>
							);
						})
					}
				</div>
			</div>
		);
	}
}

ItemsList.propTypes = {
	items: PropTypes.array
};

export default ItemsList;