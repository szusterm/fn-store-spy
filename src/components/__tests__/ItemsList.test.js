import React from 'react';
import {shallow} from 'enzyme';

import {ItemsList} from '../ItemsList/ItemsList';

const setup = (propOverrides) => {
	const props = Object.assign({
		items: [{
			name: 'Super Axe',
			price: '2000',
			rarity: 'uncommon',
			type: 'axe',
			imageSrc: 'https://example.com'
		}]
	}, propOverrides);

	const wrapper = shallow(<ItemsList {...props}/>);

	return {props, wrapper};
};

describe('ItemsList Component', () => {
	it('renders correctly', () => {
		const {wrapper} = setup();

		expect(wrapper).toMatchSnapshot();
	});

	it('shows list of ItemCards from prop', () => {
		const {wrapper, props} = setup({
			items: [{}, {}, {}]
		});

		const items = wrapper.find('ItemCard');

		expect(items.length).toBe(props.items.length);
	});

	it('passes props to every ItemCard', () => {
		const {wrapper, props} = setup({
			items: [
				{
					name: 'Super Axe',
					price: '2000',
					rarity: 'uncommon',
					type: 'axe',
					imageSrc: 'https://example.com'
				},
				{
					name: 'Something',
					price: '800',
					rarity: 'rare',
					type: 'boots',
					imageSrc: 'https://example.pl'
				}
			]
		});

		const firstItemCard = wrapper.find('ItemCard').at(0);
		const secondItemCard = wrapper.find('ItemCard').at(1);

		expect(firstItemCard.props()).toEqual(props.items[0]);
		expect(secondItemCard.props()).toEqual(props.items[1]);
	});
});