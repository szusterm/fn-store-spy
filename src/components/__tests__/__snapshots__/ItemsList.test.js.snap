// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsList Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemsList
    items={
      Array [
        Object {
          "imageSrc": "https://example.com",
          "name": "Super Axe",
          "price": "2000",
          "rarity": "uncommon",
          "type": "axe",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row no-gutter"
      >
        <div
          className="col-6 col-sm-6 col-md-4 col-lg-3 col-xl-3"
        >
          <ItemCard
            imageSrc="https://example.com"
            name="Super Axe"
            price="2000"
            rarity="uncommon"
            type="axe"
          />
        </div>
      </div>,
      "className": "container-fluid",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-6 col-sm-6 col-md-4 col-lg-3 col-xl-3"
          >
            <ItemCard
              imageSrc="https://example.com"
              name="Super Axe"
              price="2000"
              rarity="uncommon"
              type="axe"
            />
          </div>,
        ],
        "className": "row no-gutter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ItemCard
              imageSrc="https://example.com"
              name="Super Axe"
              price="2000"
              rarity="uncommon"
              type="axe"
            />,
            "className": "col-6 col-sm-6 col-md-4 col-lg-3 col-xl-3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "imageSrc": "https://example.com",
              "name": "Super Axe",
              "price": "2000",
              "rarity": "uncommon",
              "type": "axe",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row no-gutter"
        >
          <div
            className="col-6 col-sm-6 col-md-4 col-lg-3 col-xl-3"
          >
            <ItemCard
              imageSrc="https://example.com"
              name="Super Axe"
              price="2000"
              rarity="uncommon"
              type="axe"
            />
          </div>
        </div>,
        "className": "container-fluid",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-6 col-sm-6 col-md-4 col-lg-3 col-xl-3"
            >
              <ItemCard
                imageSrc="https://example.com"
                name="Super Axe"
                price="2000"
                rarity="uncommon"
                type="axe"
              />
            </div>,
          ],
          "className": "row no-gutter",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ItemCard
                imageSrc="https://example.com"
                name="Super Axe"
                price="2000"
                rarity="uncommon"
                type="axe"
              />,
              "className": "col-6 col-sm-6 col-md-4 col-lg-3 col-xl-3",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "imageSrc": "https://example.com",
                "name": "Super Axe",
                "price": "2000",
                "rarity": "uncommon",
                "type": "axe",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
